packages/core/admin/admin/src/content-manager/components/DynamicTable/BulkActionsBar/index.js

```diff
@@ -240,16 +216,42 @@ const BulkActionsBar = ({
     }
   };
 
-  const handleConfirmUnpublishAll = async () => {
+  const handleConfirmDeleteAll = async () => {
     try {
       setIsConfirmButtonLoading(true);
-      await onConfirmUnpublishAll(selectedEntries);
+      await onConfirmDeleteAll(selectedEntries);
       clearSelectedEntries();
       setIsConfirmButtonLoading(false);
     } catch (err) {
       setIsConfirmButtonLoading(false);
     }
-    handleToggleShowUnpublishAllModal();
+    handleToggleShowDeleteAllModal();
+  };
+
+  /**
+   *
+   * @param {'publish' | 'unpublish'} actionType - The bulk action to perform
+   */
+  const handleBulkAction = (actionType) => {
+    if (actionType !== 'publish' && actionType !== 'unpublish') {
+      throw new Error('The actionType must be either publish or unpublish');
+    }
+
+    const mutateActionType = actionType === 'publish' ? onConfirmPublishAll : onConfirmUnpublishAll;
+
+    setIsConfirmButtonLoading(true);
+
+    mutateActionType(selectedEntries, {
+      handleSuccess() {
+        clearSelectedEntries();
+        setIsConfirmButtonLoading(false);
+        handleToggleShowPublishAllModal();
+      },
+      handleError() {
+        setIsConfirmButtonLoading(false);
+        handleToggleShowPublishAllModal();
+      },
+    });
```

Comment:
🤷‍♂️ Let me know what y'all think, I can revert if you prefer it the other way. I just thought it was a lot of the same code. 

I'm also wondering if we should use react-query for the data fetching and bulkDelete action as well in the ListView 🤔 maybe in another PR though

Reply:
but I think we should be handling the states in the mutation callbacks, not in a try catch

Reply:
Actually I created a bug, since now calling handleToggleShowPublishAllModal each time. I thought it was single function for both states...